src/Spryker/Zed/Url/Business/Url/UrlCreator.php

```diff
@@ -115,7 +115,9 @@ protected function persistUrlEntity(UrlTransfer $urlTransfer)
         $urlTransfer->requireUrl();
         $urlEntity = $this->urlQueryContainer->queryUrl($urlTransfer->getUrl())->findOneOrCreate();
 
+        $id = $urlEntity->getIdUrl();
         $urlEntity->fromArray($urlTransfer->modifiedToArray());
```

Comment:
@julianzimmermann we use `modifiedToArray`, which will not take `null` for `id`. if field `id` was still modified and set to `null`, the problem is in the other code.